src/componentLibrary/Modal/Modal.tsx

```diff
@@ -0,0 +1,63 @@
+import { MutableRefObject, ReactNode, useEffect, useRef } from 'react';
+import Portal from 'parts/Portal';
+import { ModalContainer, ModalContent, ModalOverlay, CloseIcon } from './Modal.style';
+import { theme } from '../theme';
+import { useMountTransition } from 'utils/hooks/use-mount-transition';
+import { Icon, IconVariant } from '../Icon';
+interface ModalProps {
+  open: boolean;
+  onClose: () => void;
+  children: ReactNode;
+  initialFocusRef?: MutableRefObject<HTMLElement | null>;
+  closeIconVariant?: IconVariant;
```

Comment:
@peterslany similar to the above, do we actually want to make this configurable?

Reply:
@tomjeatt do you mean the icon variant or the initial focus?

Reply:
Sorry, iconVariant

Reply:
Actually, I added it there because it was one of your original requirements for the modal :smiley:  But if you now think it won't be utilized, then I'll remove it :)

Reply:
Ha sorry :) don't think I explained myself very well. No, don't think we'll need it for the modal.